### DISCUSSION TITLE
Need help in running the code

### DISCUSSION THREAD

**Discussion by OmarObaya** [thomeou/SALSA#14] [Q&A]
I need help to run the code i already extracted the salsa features but i have problems in train and evaluate even after appling the instructions in the read me file. Can some one help me please

**Comment by Peter-72** [comment]
@OmarObaya Hi Omar, What are your problems exactly?

**Comment by OmarObaya** [comment]
All my problems are related to path even though i followed all the data arrangement inside the code

**Comment by Peter-72** [comment]
@OmarObaya Solutions have been discussed in #2. If you still have problems after this, tag me again.

**Comment by OmarObaya** [comment]
> @OmarObaya Solutions have been discussed in #2. If you still have problems after this, tag me again.

Can we arrange an online meeting and help me run the code on my laptop?

**Comment by Peter-72** [comment]
> > @OmarObaya Solutions have been discussed in #2. If you still have problems after this, tag me again.
> 
> Can we arrange an online meeting and help me run the code on my laptop?

That is difficult for me. Let's walk it step by step here. Tell me, what are your errors?

**Reply by OmarObaya** [reply]
first of all the paths doesnt work with me even though i changed them according to my pc

**Comment by Peter-72** [comment]
@OmarObaya 

1. Go to `dataset/features/salsa/`. You will find there an `foa` folder, or a `mic` folder or both, depending on the configs you used when extracting salsa. 
2. Inside any of these folders, you will find another folder, which will have the same/similar name as `24000fs_512nfft_300nhop_5cond_9000fmaxdoa`.
NB: the naming of the folder depends on the configs you choose when extracting salsa; keep this folder's path in your mind
 
3. Go to `experiments/config/seld.yml`. 
4. Open the `seld.yml` and edit the `feature_root_dir` to have the path of the folder we talked about in point 2. It should be something like this: `feature_root_dir: 'dataset/features/salsa/mic/24000fs_512nfft_300nhop_5cond_9000fmaxdoa'`. 
5. Then edit the `gt_meta_root_dir` to have the value of `gt_meta_root_dir: 'dataset/data'`

**Reply by OmarObaya** [reply]
I adjusted the paths now i am getting an error in method called sort_tracks which is "numpy.float64 cant be converted to integer".i tried to typecast or use astype but still didnt work
